### PR TITLE
Release candidate 3.4.0

### LINKED ISSUE #200: Warnings while running tests (Ruby < 2.6)
```
Loaded suite /home/runner/work/caxlsx/caxlsx/vendor/bundle/ruby/2.5.0/gems/rake-13.0.6/lib/rake/rake_test_loader
/home/runner/work/caxlsx/caxlsx/lib/axlsx.rb:209: warning: instance variable @escape_formulas not initialized
/home/runner/work/caxlsx/caxlsx/lib/axlsx/workbook/worksheet/worksheet.rb:24: warning: instance variable @escape_formulas not initialized
Started
/home/runner/work/caxlsx/caxlsx/lib/axlsx.rb:20[9](https://github.com/caxlsx/caxlsx/actions/runs/4680319274/jobs/8291511494#step:4:10): warning: instance variable @escape_formulas not initialized
/home/runner/work/caxlsx/caxlsx/lib/axlsx/workbook/worksheet/worksheet.rb:24: warning: instance variable @escape_formulas not initialized
/home/runner/work/caxlsx/caxlsx/lib/axlsx.rb:209: warning: instance variable @escape_formulas not initialized
```

/cc @johnnyshields

### LINKED ISSUE #201: Please consider to revert (or reimplement) parts of #177
- The feature broke compatibility with Ruby 2.3 and 2.4
- Tests are performing a real network connection to `via.placeholder.com`

```
Error: test_to_xml_has_correct_r_id(TestPic): SocketError: Failed to open TCP connection to via.placeholder.com:443 (getaddrinfo: nodename nor servname provided, or not known)
```

Please also consider to disable network connection in specs to prevent this in the future. I also suggest to use `example` domains to prevent abuse and information disclosure

/cc @kiskoza

### PR DESCRIPTION
**Pull request by kiskoza**
### Description

Addressing a few issues after merging ready PRs and bumping the version to the next minor.

Close #201
Close #200 

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I added an entry to the [changelog](../blob/master/CHANGELOG.md).

### CODE REVIEW

File: lib/axlsx/util/mime_type_utils.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@ def self.get_mime_type(v)
     # @param [String] v URI
     # @return [String] File mime type
     def self.get_mime_type_from_uri(v)
-      Marcel::MimeType.for(URI.open(v))
+      if RUBY_VERSION >= '2.5'
```
Comment (tagliala) [inline lib/axlsx/util/mime_type_utils.rb:17]: I would go for `URI.respond_to?(:open)`

Comment (kiskoza) [reply lib/axlsx/util/mime_type_utils.rb]: could work for me :+1:

**Comment by kiskoza** [comment]
@tagliala can I ask for a review on this one? if I'm not mistaken we're ready to release 3.4.0 right after merging in this PR

File: lib/axlsx.rb [hunk 2]
```diff
@@ -206,7 +206,7 @@ def self.trust_input=(trust_me)
   # See https://www.owasp.org/index.php/CSV_Injection for details.
   # @return [Boolean]
   def self.escape_formulas
-    @escape_formulas.nil? ? false : @escape_formulas
+    !defined?(@escape_formulas) || @escape_formulas.nil? ? false : @escape_formulas
```
Comment (johnnyshields) [inline lib/axlsx.rb:209]: This condition isn't necessary. Should remove this change.

**Comment by johnnyshields** [comment]
This looks fine to me aside from comment above.

File: lib/axlsx.rb (hunk 2 above)
Comment (kiskoza) [reply lib/axlsx.rb:209]: this one is causing some of the warnings mentioned in #200

Comment (johnnyshields) [reply lib/axlsx.rb:209]: Hmm. Ok. Fine then. That warning is annoying :)

**Pull request merged**